pkg/queue/queue.go

```diff
@@ -118,6 +118,27 @@ func (q *Queue[T]) PullBlocking(waitable concurrency.Waitable) T {
 	return item
 }
 
+// Seq returns a iterator function that yields items from the queue as they become available.
+// The iterator will continue until the provided waitable signals done and never return empty element
+func (q *Queue[T]) Seq(waitable concurrency.Waitable) func(yield func(T) bool) {
+	return func(yield func(T) bool) {
+		for {
+			select {
+			case <-waitable.Done():
+				return
+			case <-q.notEmptySignal.Done():
+				item := q.Pull()
+				// Check if we got a valid item (not zero value)
+				if item != *new(T) {
+					if !yield(item) {
+						return
+					}
+				}
```

Comment:
Hm, this zero value check doesn't look legit. Why `int(0)` for a queue of integers should not be yielded?

If a special handling for some values is needed, may be an optional filter predicate could be added as a function parameter?

Consider: https://github.com/parametalol/curry/blob/dd489b77e7765efcba70b1f9c838973c2813960f/seq/take.go#L38

```suggestion
func (q *Queue[T]) Seq(waitable concurrency.Waitable) func(yield func(T) bool) {
	var zeroValue T
	return func(yield func(T) bool) {
		for {
			select {
			case <-waitable.Done():
				return
			case <-q.notEmptySignal.Done():
				// Check if we got a valid item (not zero value)
				if item := q.Pull(); item != zeroValue && !yield(item) {
					return
				}
```

Reply:
Solved by extracting `pull` that returns if queue was empty